interfaces/seccomp/backend.go

```diff
@@ -162,6 +163,69 @@ func bpfBinPath(srcName string) string {
 	return filepath.Join(dirs.SnapSeccompDir, strings.TrimSuffix(srcName, ".src")+".bin")
 }
 
+func parallelCompile(compiler Compiler, profiles []string) error {
+	if len(profiles) == 0 {
+		// no profiles, nothing to do
+		return nil
+	}
+
+	profilesQueue := make(chan string, len(profiles))
+	res := make(chan error, len(profiles))
+	cpus := runtime.NumCPU()
+
+	var wg sync.WaitGroup
+	wg.Add(cpus)
+
+	// launch as many workers as we have CPUs
+	for i := 0; i < cpus; i++ {
+		go func() {
+			defer wg.Done()
+			for {
+				profile, ok := <-profilesQueue
+				if !ok {
+					break
+				}
+				in := bpfSrcPath(profile)
+				out := bpfBinPath(profile)
+
+				// snap-seccomp uses AtomicWriteFile internally, on failure the
+				// output file is unlinked
+				if err := compiler.Compile(in, out); err != nil {
+					res <- fmt.Errorf("cannot compile %s: %v", in, err)
+				} else {
+					res <- nil
+				}
+			}
+		}()
+	}
+
+	for _, p := range profiles {
+		profilesQueue <- p
+	}
+	close(profilesQueue)
+	wg.Wait()
+	close(res)
+
+	var firstErr error
+	for err := range res {
```

Comment:
no strong opinion but instead of setting up a WaitGroup and close res we could just count up to len(profiles) here. With that approach res doesn't need to be len(profiles) big, could just be N*2 big.  With N as defined above.